tests/subgraph/pbab-mapping/pbab-mapping.test.ts

```diff
@@ -1842,6 +1862,306 @@ test("GenArt721Core2PBAB: Can handle transfer", () => {
+  ];
+
+  // add event
+  handleExternalAssetDependencyUpdated(event);
+  // checks project external asset dependency count
+  assert.fieldEquals(
+    PROJECT_ENTITY_TYPE,
+    fullProjectId,
+    "externalAssetDependencyCount",
+    _externalAssetDependencyCount0.toString()
+  );
+  // checks project's updatedAt 
+  assert.fieldEquals(
+    PROJECT_ENTITY_TYPE,
+    fullProjectId,
+    "updatedAt",
+    CURRENT_BLOCK_TIMESTAMP.toString()
+  );
+  // checks project external asset dependency cid
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + '-' + _index0.toString(),
+    "cid",
+    IPFS_CID
+  );
+  // checks project external asset dependency dependency type
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + '-' + _index0.toString(),
+    "dependencyType",
+    FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[_dependencyType0.toI32()]
+  );
+  // checks project external asset dependency project relationship
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + '-' + _index0.toString(),
+    "project",
+    fullProjectId
+  );
+  
+  const updateEvent: ExternalAssetDependencyUpdated = changetype<ExternalAssetDependencyUpdated>(newMockEvent());
+  updateEvent.address = TEST_CONTRACT_ADDRESS;
+  updateEvent.block.timestamp = CURRENT_BLOCK_TIMESTAMP;
+
+  const _dependencyType1 = BigInt.fromI32(1);
+  const _externalAssetDependencyCount1 = BigInt.fromI32(2);
+  updateEvent.parameters = [
+    new ethereum.EventParam("_projectId", ethereum.Value.fromUnsignedBigInt(projectId)),
+    new ethereum.EventParam("_index", ethereum.Value.fromUnsignedBigInt(_index0)),
+    new ethereum.EventParam("_cid", ethereum.Value.fromString(IPFS_CID2)),
+    new ethereum.EventParam("_dependencyType", ethereum.Value.fromUnsignedBigInt(_dependencyType1)),
+    new ethereum.EventParam("_externalAssetDependencyCount", ethereum.Value.fromUnsignedBigInt(_externalAssetDependencyCount1))
+  ];
+  handleExternalAssetDependencyUpdated(updateEvent);
+
+  // checks project external asset dependency cid
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + '-' + _index0.toString(),
+    "cid",
+    IPFS_CID2
+  );
+  // checks project external asset dependency dependency type
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + '-' + _index0.toString(),
+    "dependencyType",
+    FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[_dependencyType1.toI32()]
+  );
+
+  // checks project external asset dependency count
+  assert.fieldEquals(
+    PROJECT_ENTITY_TYPE,
+    fullProjectId,
+    "externalAssetDependencyCount",
+    _externalAssetDependencyCount1.toString()
+  );
+  
+});
+
+test("GenArt721Core2EngineFlex: Can remove a project external asset dependency", () => {
```

Comment:
I think adding another test that confirms e.g. removing external asset at index 0 after adding two external assets, then using `assert.entityCount` to confirm there are no duplicate entities in store, would be 🔥

Reply:
sounds good, but couldn't that assertion be preformed with this test as well/not sure why we would need a new separate test?

Reply:
EDIT: I actually got confused here with the contract tests, where I already do explicitly test removal after multiple adds. This test is currently only adding one and removing it. I still think we can use a single test here, will just update this current one to add more than one dep!

Reply:
added 👍